GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -15,13 +16,10 @@ public void FlushFileBuffers(string path)
 
         public void MoveAndOverwriteFile(string sourceFileName, string destinationFilename)
         {
-            // TODO(Mac): Use native API
-            if (File.Exists(destinationFilename))
+            if (Rename(sourceFileName, destinationFilename) != 0)
             {
-                File.Delete(destinationFilename);
+                NativeMethods.ThrowLastWin32Exception($"Failed to renname {sourceFileName} to {destinationFilename}");

```

Comment:
A little weird throwing a Win32Exception from the Mac code but in searching it looks like we are already dependent on that exception.

Reply:
Yep it is weird, but this is what the .NET Core APIs chose to do as well. We just need to translate "Win32Exception" to "NativeException".

Reply:
Yep here is an example:

https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs#L44

.NET Core's I/O methods will translate native error codes into a more appropriate exception when possible (e.g. IOException), but in our code we consistently throw Win32Exceptions for all failed native calls.